### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by vanhauser-thc**

### CODE REVIEW

**Review by hexcoder- — APPROVED** [review]

File: src/afl-common.c [hunk 1]
```diff
@@ -51,6 +55,66 @@ u8  last_intr = 0;
   #define AFL_PATH "/usr/local/lib/afl/"
 #endif
 
+u32 check_binary_signatures(u8 *fn) {
```
Comment (domenukk) [inline src/afl-common.c:58]: I (still) don't understand why we need this, the compiled target knows if it is using a deferred forkserver or not...
There should be a better way than to have the parent scan the binary, imho...

**Review by domenukk — APPROVED** [review]

File: src/afl-common.c (hunk 1 above)
Comment (vanhauser-thc) [reply src/afl-common.c:58]: yeah maybe. we can put that on the todo list and do something about this at some point. eg. add a change to a global var in afl-compiler-rt.o.c in the __AFL_INIT macro etc.

File: TODO.md [hunk 2]
```diff
@@ -1,30 +1,27 @@
 # TODO list for AFL++
 
-## Roadmap 3.00+
+## TODO
```
Comment (domenukk) [inline TODO.md]: TODO is already in the title (minor nitpick)

File: src/afl-fuzz-queue.c [hunk 3]
```diff
@@ -315,7 +315,96 @@ void mark_as_redundant(afl_state_t *afl, struct queue_entry *q, u8 state) {
...
+/* check if pointer is ascii or UTF-8 */
+
+u8 check_if_text_buf(u8 *buf, u32 len) {
+
+  u32 offset = 0, ascii = 0, utf8 = 0;
+
+  while (offset < len) {
+
+    // ASCII: <= 0x7F to allow ASCII control characters
+    if ((buf[offset + 0] == 0x09 || buf[offset + 0] == 0x0A ||
```
Comment (domenukk) [inline src/afl-fuzz-queue.c:327]: why `+ 0` here?

**Review by domenukk — APPROVED** [review]

File: src/afl-fuzz-queue.c (hunk 3 above)
Comment (vanhauser-thc) [reply src/afl-fuzz-queue.c:327]: just for copy-paste-ability + readability

**Pull request merged**